### PR TITLE
GH-9: Fixed type errors.

### LINKED ISSUE #9: Bogus error handling
I had this error with `@1.0.1`:

```
Run shallwefootball/s3-upload-action@v1.0.1
/home/runner/work/_actions/shallwefootball/s3-upload-action/v1.0.1/dist/index.js:8307
            throw err;
            ^

Error [TypeError]: (s || "").replace is not a function
    at escapeData (/home/runner/work/_actions/shallwefootball/s3-upload-action/v1.0.1/dist/index.js:8121:10)
    at Command.toString (/home/runner/work/_actions/shallwefootball/s3-upload-action/v1.0.1/dist/index.js:8115:35)
    at issueCommand (/home/runner/work/_actions/shallwefootball/s3-upload-action/v1.0.1/dist/index.js:8078:30)
    at Object.issue (/home/runner/work/_actions/shallwefootball/s3-upload-action/v1.0.1/dist/index.js:8082:5)
    at Object.error (/home/runner/work/_actions/shallwefootball/s3-upload-action/v1.0.1/dist/index.js:9545:15)
    at ManagedUpload.callback (/home/runner/work/_actions/shallwefootball/s3-upload-action/v1.0.1/dist/index.js:2430:21)
    at ManagedUpload.cleanup (/home/runner/work/_actions/shallwefootball/s3-upload-action/v1.0.1/dist/index.js:20775:10)
    at Request.<anonymous> (/home/runner/work/_actions/shallwefootball/s3-upload-action/v1.0.1/dist/index.js:20654:16)
    at Request.callListeners (/home/runner/work/_actions/shallwefootball/s3-upload-action/v1.0.1/dist/index.js:12939:20)
    at Request.emit (/home/runner/work/_actions/shallwefootball/s3-upload-action/v1.0.1/dist/index.js:12911:10) {
  message: '(s || "").replace is not a function',
  code: 'TypeError',
  time: 2020-07-02T16:11:58.172Z
}
```

Could not figure out what this is, then noticed the incorrect typing here:

https://github.com/shallwefootball/upload-s3-action/blob/7ca30b18d2c7fa3f9c51e92216317718adf41c4e/index.js#L37

and here:

https://github.com/shallwefootball/upload-s3-action/blob/7ca30b18d2c7fa3f9c51e92216317718adf41c4e/index.js#L66

Obviously, my upload has failed but I do not know why because the error handling has failed. I will provide a PR soon.

### PR DESCRIPTION
**Pull request by kittaakos**
- Log the `message` of the  `Error` object.
 - Streamlined output to string for `fromjson`.
 - Aligned the documenation with the new `object_locations`value.
 - Regenerated `dist`.

Closes #9

Signed-off-by: Akos Kitta <kittaakos@typefox.io>

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Upload S3
+name: Upload S3 - Fix GH-9
```
Comment (shallwefootball) [inline action.yml:1]: I think it is unnecessary '- Fix GH-9'

remove plz

File: index.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// @ts-check
```
Comment (shallwefootball) [inline index.js:1]: remove too. 
unnecessary

File: README.md [hunk 3]
```diff
@@ -81,7 +81,7 @@ The following settings must be passed as environment variables as shown in the e
...
 ## Action outputs
 
-| name               | description                                                                             |
-| ------------------ | --------------------------------------------------------------------------------------- |
-| `object_key`       | Uploaded object key generated by [shortid](https://github.com/dylang/shortid) in Bucket |
-| `object_locations` | Object Locations                                                                        |
+| name               | description                                                                                   |
+| ------------------ | --------------------------------------------------------------------------------------------- |
+| `object_key`       | Uploaded object key generated by [shortid](https://github.com/dylang/shortid) in Bucket       |
+| `object_locations` | The URLs of the uploaded objects as a JSON string. E.g: `["location/file1","location/file2"]` |
```
Comment (shallwefootball) [inline README.md:87]: why use the array to string?? 

this is intended to control by pragmatically in other actions

Comment (shallwefootball) [reply README.md:87]: @kittaakos you're right. 
must have to stringify Object. 

but core was update this https://github.com/actions/toolkit/issues/370

so could you update @actions/core?

**Comment by kittaakos** [comment]
@shallwefootball, sorry for the delay and messing up with the PRs. I opened a follow-up with a bare minimum changeset. Here is the follow-up: https://github.com/shallwefootball/upload-s3-action/pull/17

**Pull request closed without merging**